javalib/src/main/scala/java/util/concurrent/CopyOnWriteArrayList.scala

```diff
@@ -0,0 +1,348 @@
+package java.util
+package concurrent
+
+import java.lang.{reflect => jlr}
+
+import scala.annotation.tailrec
+import scala.collection.JavaConversions._
+
+import scala.scalajs._
+
+class CopyOnWriteArrayList[E](private var inner: js.Array[E])
+    extends AbstractCopyOnWriteArrayList[E] with List[E] with RandomAccess
+    with Cloneable with Serializable { self =>
+
+  // requiresCopyOnWrite is false if and only if no other object (like the iterator)
+  // may have a reference to inner
+  private var requiresCopyOnWrite = false
+
+  def this() = {
+    this(new js.Array[E])
```

Comment:
Any reason this is not the primary constructor? It seems all the other constructors call this one anyway.